src/accounts/context/userAccount.tsx

```diff
@@ -73,7 +94,24 @@ export const UserAccountProvider: FC<Props> = React.memo(({children}) => {
     } catch (error) {
       event('multiAccount.retrieveAccounts.error', {error})
     }
-  }, [dispatch])
+  }, [dispatch, defaultAccountId])
+
+  async function handleSetDefaultAccount(newDefaultAcctId) {
+    const accountName = getAccountNameById(newDefaultAcctId)
+
+    try {
+      const resp = await putAccountsDefault({data: {id: newDefaultAcctId}})
+      setDefaultAccountId(newDefaultAcctId)
+
+      if (resp.status !== 204) {
+        dispatch(notify(accountDefaultSettingError(accountName)))
+      } else {
+        dispatch(notify(accountDefaultSettingSuccess(accountName)))
+      }
+    } catch (error) {
+      dispatch(notify(accountDefaultSettingError(accountName)))
+    }
+  }
```

Comment:
i'm just going to repeat my previous concerns that this being in context turns an api-level operation that operates across the wire into something that react, the ui layer of this app, is aware of. the code in this context area should be refactored to make use of the existing app architecture and separation of concerns we already have.

Reply:
noted